src/components/Article.jsx

```diff
@@ -21,24 +21,37 @@ const Article = () => {
     })
   }, [id]);
 
+  const handleVote = (voteAdjustment) => {
+    setArticle((article) => ({
+      ...article,
+      votes: article.votes + voteAdjustment,
+    }));
+
+    voteOnArticle(id, voteAdjustment)
+      .then((updatedArticle) => {
+        setArticle(updatedArticle);
+      })
+    }
```

Comment:
The optimistic rendering works great, but when refreshing the page, the votes are back to what they were. This is because your patch request failed.